hazelcast/src/main/java/com/hazelcast/spi/OperationFactory.java

```diff
@@ -26,9 +26,8 @@
  * {@link OperationService} uses this factory to create {@link Operation}s for each partition by calling
  * {@link #createOperation()}
  * </p>
- *
  */
-public interface OperationFactory extends DataSerializable {
```

Comment:
"[...] Otherwise we're stucked with the package and naming forever." <-- I don't see how. 

It's only important for the actual factory implementations to implement the `IdentifiedDataSerializable` interface. Serilization strategy declared by the interface is good as a reminder / way to force the implementer to use a specific strategy. However I don't see any impact on Rolling Upgrade.